gateway/src/apicast/threescale_utils.lua

```diff
@@ -152,22 +152,34 @@ function _M.connect_redis(options)
 
   local ok, err = red:connect(_M.resolve(host, port))
   if not ok then
-    return nil, _M.error("failed to connect to redis on ", host, ":", port, ": ", err)
+    if not resilient then
```

Comment:
@abarrak I think it's better to add this as another condition rather than a nested if statement. `if not ok and if not resilient then` because we only want to execute something when both return false or both return true. It's not super important and maybe @eloycoto might be able to comment whether the nested if statement could introduce any additional unwanted processing.

Reply:
And what would you suggest for the other case?
Does `elseif not ok then …` sound good enough?

Reply:
I'd just suggest something like:

```
if not ok and not resilient then
  return nil, _M.error("failed to connect to redis on ", host, ":", port, ": ", err)
else
  return nil, _M.error_silently("failed to connect to redis on ", host, ":", port, ": ", err)
end
```

This might just be about style though so let¡s wait for a comment from someone more knowledgeable in Lua than myself because if it doesn't introduce unnecessary processing then we can leave it as you have already written it. I assume this is executed on every single request though right? That's why I was a little concerned, I don't see us opening  a connection with Redis and reusing that in this code, we open a new connection for every inbound request.

Reply:
Alright will wait for @eloycoto feedback.
I considered it in the first, but changed to the current structure, due the fact that original condition was just `if not ok` only, and using `else` alone here would catch the cases where response is `ok`.

Many thanks for the review comments so far 🌹 will make sure to implement it in upcoming commit.